.github/workflows/build.yml

```diff
@@ -0,0 +1,200 @@
+      - name: Get the tag value from github
+        # see https://stackoverflow.com/a/58178121/1138710
+        id: vars
+        run: echo ::set-output name=tag::${GITHUB_REF#refs/*/}
+      - name: Save git version so the app can show it to users
+        env:
+          # we need this hack because github default env vars don't include the tag itself
+          GIT_TAG: ${{ steps.vars.outputs.tag }}
+        run: sed -ri "s/COMMIT/$GITHUB_SHA/" debugInfo.json; sed -ri "s/TAG/$GIT_TAG/" debugInfo.json
+      - name: Jetify react files
+        run: yarn jetifier
+      - name: Increase filesystem inotify watch limit
+        run: sudo sysctl fs.inotify.max_user_watches=524288
+      - name: Build dev APK
+        env:
+          ORG_GRADLE_PROJECT_MAPSWIPE_RELEASE_KEY_ALIAS: ${{ secrets.ORG_GRADLE_PROJECT_MAPSWIPE_RELEASE_KEY_ALIAS }}
+          ORG_GRADLE_PROJECT_MAPSWIPE_RELEASE_KEY_PASSWORD: ${{ secrets.ORG_GRADLE_PROJECT_MAPSWIPE_RELEASE_KEY_PASSWORD }}
+          ORG_GRADLE_PROJECT_MAPSWIPE_RELEASE_STORE_FILE: ${{ secrets.ORG_GRADLE_PROJECT_MAPSWIPE_RELEASE_STORE_FILE }}
+          ORG_GRADLE_PROJECT_MAPSWIPE_RELEASE_STORE_PASSWORD: ${{ secrets.ORG_GRADLE_PROJECT_MAPSWIPE_RELEASE_STORE_PASSWORD }}
+        run: yarn buildDev
+      - name: Build production APK
+        env:
+          ORG_GRADLE_PROJECT_MAPSWIPE_RELEASE_KEY_ALIAS: ${{ secrets.ORG_GRADLE_PROJECT_MAPSWIPE_RELEASE_KEY_ALIAS }}
+          ORG_GRADLE_PROJECT_MAPSWIPE_RELEASE_KEY_PASSWORD: ${{ secrets.ORG_GRADLE_PROJECT_MAPSWIPE_RELEASE_KEY_PASSWORD }}
+          ORG_GRADLE_PROJECT_MAPSWIPE_RELEASE_STORE_FILE: ${{ secrets.ORG_GRADLE_PROJECT_MAPSWIPE_RELEASE_STORE_FILE }}
+          ORG_GRADLE_PROJECT_MAPSWIPE_RELEASE_STORE_PASSWORD: ${{ secrets.ORG_GRADLE_PROJECT_MAPSWIPE_RELEASE_STORE_PASSWORD }}
+        run: yarn buildProd
+      - name: Upload APK to Github releases (beta build)
+        uses: ncipollo/release-action@v1
+        if: startsWith(github.ref, 'refs/tags/') && endsWith(github.ref, '-beta')
+        with:
+          artifacts: "./app/build/outputs/apk/dev/release/app-dev-release.apk"
+          # mark it as prerelease as it's a beta/dev build
+          prerelease: true
+          token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Upload APK to Github releases (production build)
+        uses: ncipollo/release-action@v1
+        if: startsWith(github.ref, 'refs/tags/') && !endsWith(github.ref, '-beta')
+        with:
+          artifacts: "./app/build/outputs/apk/production/release/app-production-release.apk"
+          prerelease: true
+          token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Release to PlayStore
+        uses: maierj/fastlane-action@v2.0.1
+        # only deploy a release on a non-beta tag
+        # FIXME: there does not seem to be a way with github actions (as of may 2021) to get
+        # both tag and branch, so for now we will rely on the user not tagging randomly
```

Comment:
hey,
this looks interesting. So basically we could do a production release from any branch and it will be released to playstore.

What does this mean?

When it's released to playstore there is still manual approval needed, to make it available to users? So even if somebody would accidentally release a production version this would not be a problem, since we could just delete it in playstore?

Reply:
I'm adding a check in the `scripts/version.sh` script to select `beta` or `production` based on the current branch. Obviously this does not prevent manual tagging which would throw things off, but as you say, we still have a manual step in the app stores after this, and there is only a limited number of people who can actually trigger this job, so I think overall the risk is low.